lib/os/fdtable.c

```diff
@@ -227,6 +227,12 @@ int ioctl(int fd, unsigned long request, ...)
 	return res;
 }
 
+/*
+ * In the SimpleLink case, we have yet to add support for the fdtable
+ * feature. The socket offload subsys has already defined fcntl, hence we
+ * avoid redefining fcntl here.
+ */
+#ifndef CONFIG_SOC_FAMILY_TISIMPLELINK
```

Comment:
This is misplaced here. This file is not SoC-specific, so should not contain stuff like that. I'm going to approve this PR, but please work towards cc32xx compatibility with Zephyr POSIX subsys (this #ifdef effectively disables it), and implementation of fcntl via fdtable->ioctl().

Reply:
I very much agree with @pfalcon here. This is just in wrong place.
Aren't you installing `simplelink_fcntl` function in simplelink_sockets.c so what is the purpose of this stuff here?

Reply:
@pfalcon Yes I agree that ultimately for cc32xx to conform with the flow of using fdtable is probably the right thing to do in the long run. This is only done as a short term solution for fcntl support in 1.14.

@jukkar I had to resort to taking this code out for SimpleLink because there needs to be another definition of fcntl() in subsys/net/lib/sockets/socket_offload.c which generates the energy to call simplelink_fcntl(). Originally I wanted to do this for all cases where NET_SOCKET_OFFLOAD is set, but to your earlier point that'd prevent others who have enabled fdtable support in offloaded sockets to benefit from unified file descriptor handling. So given where we are with SimpleLink today (ie. no fdtable support) and what was agreed upon in #11891 , this made the most sense to me in order to prevent fcntl() from being defined twice, and to be able to continue to follow a separate path for all offloaded functions.

Reply:
@vanti Ok, I suppose we can leave it like this until a proper support is added.